public/app/core/logs_model.ts

```diff
@@ -198,10 +198,11 @@ function isLogsData(series: DataFrame) {
 export function dataFrameToLogsModel(
   dataFrame: DataFrame[],
   intervalMs: number | undefined,
-  timeZone: TimeZone
+  timeZone: TimeZone,
+  isLoki?: boolean
 ): LogsModel {
```

Comment:
What about passing the datasource instead of `isLoki`? It would give us more reusability if there are more specific processing.

Reply:
Good point, changed.